estate/models/estate_property.py

```diff
@@ -0,0 +1,96 @@
+    selling_price = fields.Float(readonly=True, copy=False)
+    best_offer = fields.Float(compute='_compute_best_offer')
+    sales_person_id = fields.Many2one('res.users', string='Salesperson', default=lambda self: self.env.user)
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string='Offer')
+    buyer_id = fields.Many2one('res.partner', string='Customer')
+    bedrooms = fields.Integer(default=2)
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_orientation = fields.Selection(
+        string='Orientation',
+        selection=[('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West')],
+    )
+    living_area = fields.Integer()
+    garden_area = fields.Integer()
+    total_area = fields.Float(compute='_compute_total_area')
+    state = fields.Selection(
+        string='State',
+        selection=[
+            ('new', 'New'),
+            ('offer received', 'Offer Received'),
+            ('offer accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled'),
+        ],
+        default='new',
+        readonly=True,
+    )
+
+    _sql_constraints = [
+        ('expected_price', 'CHECK(expected_price >= 0 )', 'A price should always be possitive'),
+        ('selling_price', 'CHECK(selling_price >= 0 )', 'A price should always be possitive'),
+    ]
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.onchange('offer_ids')
+    def _onchange_status(self):
+        for record in self:
+            if record.state not in ['offer accepted', 'sold', 'cancelled']:
+                all_status = record.offer_ids.mapped('status')
+                new_state = 'new'
+                if all_status:
+                    new_state = 'offer received'
+            record.state = new_state
+
+    @api.depends('offer_ids')
+    def _compute_best_offer(self):
+        for record in self:
+            all_price = record.offer_ids.mapped('price')
+            if all_price:
+                record.best_offer = max(all_price)
+            else:
+                record.best_offer = 0
+
+    @api.onchange('garden')
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = ''
+
+    def action_property_sold(self):
+        for record in self:
+            if record.state == 'cancelled':
+                raise exceptions.UserError('property already cancelled')
+            record.state = 'sold'
+
+    def action_property_cancelled(self):
+        for record in self:
+            if record.state == 'sold':
+                raise exceptions.UserError('property already sold')
```

Comment:
```suggestion
                raise UserError(self.env._('property already sold'))
```
As Séna would say:

> In Odoo, translatable strings in "data"-type content are automatically exported for translation. However, other user-facing strings, Odoo cannot automatically know whether they are translatable. To ensure these strings are exported for translation, you need to explicitly mark them. This is achieved by wrapping the literal string with the self.env._() function. The _() function signals to Odoo's translation system that the string should be included in the translation catalog.
> 
> More info here: https://www.odoo.com/documentation/18.0/developer/howtos/translations.html#explicit-exports
> 

You can check this line here:
https://github.com/odoo/odoo/blob/05cff3b7d866f6bc95c4b32f343ae14a4da946f2/addons/account/models/account_move.py#L2075